UiPath.FreeRdpClient/UiPath.SessionTools/WTS/Wts.InformationProvider.cs

```diff
@@ -0,0 +1,67 @@
+﻿using Nito.Disposables;
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+using Windows.Win32;
+using Windows.Win32.System.RemoteDesktop;
+
+namespace UiPath.SessionTools;
+
+partial class Wts
+{
+    public record InfoProvider(int SessionId)
+    {
+        public virtual IDisposable? Query(WTS_INFO_CLASS infoClass, out IntPtr pInfo)
```

Comment:
I don't think this needs to be public, not even protected
[these](https://learn.microsoft.com/en-gb/windows/win32/api/wtsapi32/ne-wtsapi32-wts_info_class#members) are the only possibilities of using it